### PR TITLE
libguestfs: 1.38.6 -> 1.40.2

### PR DESCRIPTION
**Pull request by Amar1729**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc

Note: While the [homepage](http://libguestfs.org/) says current stable is 1.39.11, the [downloads page](http://download.libguestfs.org/) indicates stable is 1.40.

### CODE REVIEW

**Comment by vdemeester** [comment]
@GrahamcOfBorg build libguestfs

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build libguestfs

File: pkgs/development/libraries/libguestfs/default.nix [hunk 1]
```diff
@@ -65,6 +65,8 @@ stdenv.mkDerivation rec {
 
   doInstallCheck = appliance != null;
   installCheckPhase = ''
+    runHook preInstallCheck
```
Comment (volth) [inline pkgs/development/libraries/libguestfs/default.nix]: please stop inserting needless `runHook`s

File: pkgs/development/libraries/libguestfs/default.nix [hunk 2]
```diff
@@ -77,12 +79,14 @@ stdenv.mkDerivation rec {
     mkfs ext2 /dev/sda1
     list-filesystems
     EOF
+
+    runHook postInstallCheck
```
Comment (volth) [inline pkgs/development/libraries/libguestfs/default.nix]: please stop inserting needless `runHook`s

File: pkgs/development/libraries/libguestfs/default.nix (hunk 1 above)
Comment (c0bw3b) [reply pkgs/development/libraries/libguestfs/default.nix]: I don't think they are needless. Phases overrides ignoring pre and post hooks makes for bad examples. I'm not waiting for the situation when we'll need to add a pre build task on a batch of derivations for whatever reason and then realise half of them are just ignoring it. (it's an example).
So I believe it's good hygiene to start and end with pre and post hook whenever one resort to rewriting an entire phase.

**Review by c0bw3b — APPROVED** [review]
Result of `nix-review pr 67572` [1](https://github.com/Mic92/nix-review)
<details>
  <summary>6 package were build:</summary>

  - libguestfs
  - libguestfs-with-appliance
  - python27Packages.guestfs
  - python37Packages.guestfs
  - python38Packages.guestfs
  - vagrant
</details>

File: pkgs/development/libraries/libguestfs/default.nix (hunk 1 above)
Comment (volth) [reply pkgs/development/libraries/libguestfs/default.nix]: No, it is not.
The almost only place when they are needed are function like `buildPerlModule` which accept `args` and pass this args down
https://github.com/NixOS/nixpkgs/blob/c28659e2fbe37841bf2528a79759d13bf6ce87ef/pkgs/top-level/perl-packages.nix#L45-L51

Without explicit runHook, the hooks will be lost and `buildPerlModule { preSometing = "..." }` won't work.


About overrides I have three objection:
1. overrides is an exceptional mechanism which should not require explicit support. especially which requires adding about 200.000 lines if your proposal will be implementing treewide.
2. If it really needed, it should be adding it treewide, adding it using a partisan tactic in randomly chosen dozen places is ridiculous.
3. in my opinion, hook-support are is not even needed for overrides as in override if one want to prepend/postpend code she can write
```
somederivation.overrideDerivation(old: {
installPhase = ''
  some prefix
  ${old.installPhase}
  some postfix
'';
}
```
explicit hooks only add confusions

**Pull request merged**